pkg/machines/components/create-vm-dialog/pxe-helpers.js

```diff
@@ -0,0 +1,187 @@
+
+/**
+ * Returns a list of all virbrX Virtual Networks.
+ * @param {array} virtualNetworks - An array of object containing Virtual Networks.
+ */
+function getLibvirtNetworkBridges(virtualNetworks) {
+    return virtualNetworks
+            .filter(network => network.bridge)
+            .map(network => network.bridge.name);
+}
+
+/**
+ * Filters an array of node devices returning only devices of specific connection and capability.
+ * @param {array} nodeDevices - An array of object containing NodeDevices.
+ * @param {string} connectionName - The libvirt connection name.
+ * @param {string} type - The capability type, ex 'net'.
+ */
+function getNodeDevicesOfType(nodeDevices, connectionName, type) {
+    return nodeDevices.filter(nodeDevice => {
+        const capabilityType = nodeDevice.capability.type;
+
+        return (nodeDevice.connectionName == connectionName &&
+                capabilityType == type);
+    });
+}
+
+/**
+ * Return the Virtual Network matching a name from a Virtual Networks list.
+ * @param {string} virtualNetworkName.
+ * @param {array} virtualNetworks - An array of object containing Virtual Networks.
+ * @param {string} connectionName - The libvirt connection name.
+ */
+export function getVirtualNetworkByName(virtualNetworkName, virtualNetworks, connectionName) {
+    return virtualNetworks.filter(virtualNetwork =>
+        virtualNetwork.connectionName == connectionName &&
+        virtualNetwork.name == virtualNetworkName
+    )[0];
+}
+
+/**
+ * Return a short description of the Virtual Network.
+ * @param {object} virtualNetwork - A Virtual Network object.
+ */
+function getVirtualNetworkDescription(virtualNetwork) {
+    let mode, dev;
+    let forward = virtualNetwork.forward;
+
+    if (forward) {
+        mode = forward.mode;
+        dev = forward.interface && forward.interface.dev;
+    }
+
+    if (mode || dev) {
+        if (!mode || mode == 'nat') {
+            if (dev)
+                return cockpit.format('NAT to $0', dev);
+            else
+                return 'NAT';
+        } else if (mode == 'route') {
+            if (dev)
+                return cockpit.format('Route to $0', dev);
+            else
+                return 'Routed Network';
+        } else {
+            if (dev)
+                return cockpit.format('$0 to $1', mode, dev);
+            else
+                return cockpit.format('$0 network', mode.toUpperCase());
+        }
+    } else {
+        return _("Isolated network");
+    }
+}
+
+/**
+ * @param {array} nodeDevices - An array of object containing NodeDevices.
+ * @param {object} virtualNetwork - A Virtual Network object.
+ */
+export function getVirtualNetworkPXESupport(virtualNetwork) {
+    if (virtualNetwork.forward && virtualNetwork.forward.mode != 'nat') {
+        return true;
+    }
+
+    let ips = virtualNetwork.ip;
+
+    for (let i = 0; i < ips.length; i++)
+        if (ips[i].dhcp.bootp)
+            return true;
+
+    return false;
```

Comment:
This could be simplified with `return virtualNetwork.ip.find(ip => ip.dhcp.bootp)` (prepend a `!!` if returning a truthy value or `undefined` instead of `true`/`false` is not acceptable)